src/algorithms/shortestPath/getPath.ts

```diff
@@ -12,22 +19,25 @@ export function getPath<Node, LinkProps>(
   tracks: TraversingTracks<NoInfer<Node>>,
   source: NoInfer<Node>,
   destination: NoInfer<Node>,
+  weightFunction: (edgeWeight: number, currentPathWeight: number, hop: number) => number = addWeightFunction
```

Comment:
Prefer an object as a single parameter instead of multi parameters. It makes evolutions easier.

Also, make the properties more explicit + add jsdoc